.github/workflows/release.yaml

```diff
@@ -0,0 +1,32 @@
+---
+name: Release
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  semrel:
+    name: Semantic Release
+    runs-on: ubuntu-latest
+    permissions:
+      actions: none
+      checks: none
+      contents: none
+      deployments: none
+      issues: none
+      packages: write
+      pull-requests: none
+      repository-projects: none
+      security-events: none
+      statuses: none
+      id-token: none
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
does this fetch deep enough for semrel to word?

Reply:
idk, it works in other repos

Reply:
`fetch-depth` defaults to `1` to make it only fetch the last commit. this is often fine when analyzing 1 commit is enough to tag a release. if semrel needs to take multiple commit into account to figure out the new version then `fetch-depth: 0` will ensure that it has access to the full commit history.

This was for sure the case a while back and might have changed in newer versions of the checkout action. Do you know of an example repo that demonstrated it doing the right thing even when semrel doesn't have a complete picture in it's working copy?

Reply:
lets try this as is then